src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Helpers/Parser/OptionsConfigurationServiceCollectionExtensions.cs

```diff
@@ -85,9 +85,12 @@ private void RegisterMethodInvocation_ServiceCollectionExt(BinderInvocation invo
 
             private void RegisterTypeForMethodGen(MethodsToGen_Extensions_ServiceCollection overload, TypeSpec typeSpec)
             {
+                RegisterTypeForBindCoreMainGen(typeSpec);
+
                 _sourceGenSpec.MethodsToGen_ServiceCollectionExt |= overload;
                 _sourceGenSpec.Namespaces.Add("Microsoft.Extensions.DependencyInjection");
-                RegisterTypeForBindCoreMainGen(typeSpec);
+                // Emitting refs to IOptionsChangeTokenSource, ConfigurationChangeTokenSource, IConfigureOptions<>, ConfigureNamedOptions<>.
+                _sourceGenSpec.Namespaces.Add("Microsoft.Extensions.Options");
```

Comment:
Have you tested the case when the user doesn't reference the Options assembly?  I'm surprised to see this functional change here in with the test changes.  Was something broken before?

Reply:
Yes the diff fixes an RC-2 regression added in https://github.com/dotnet/runtime/pull/91359. Discovered while working on this test enhancement that's been on my list for a while. This nuance escaped me after iterating on the PR a few times. Will update template.

> Have you tested the case when the user doesn't reference the Options assembly?

An intrinsic characteristic of parsing invocation operations is that Roslyn won't parse them successfully unless the member being reference is in the compilation: https://github.com/dotnet/runtime/blob/cf3328cceecccf781afa3e1ba619cf42830ead60/src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Helpers/Parser/BinderInvocation.cs#L19. This behavior does not need to be tested here.

Reply:
I see, we're inside the OptionsConfigurationServiceCollectionExtensions so we're only using this if we know we're emitting code for an options extension.